toolkit/tools/pkg/imagecustomizerlib/extractpartitions_test.go

```diff
@@ -0,0 +1,127 @@
+package imagecustomizerlib
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddSkippableFrame(t *testing.T) {
+	// Create a skippable frame containing the metadata and prepend the frame to the partition file
+	skippableFrameMetadata, err := createSkippableFrameMetadata()
+	assert.NoError(t, err)
+
+	// Create test raw partition file
+	partitionRawFilepath, err := createTestRawPartitionFile("test.raw")
```

Comment:
This package has a [main_test.go](https://github.com/microsoft/azurelinux/blob/3.0-dev/toolkit/tools/pkg/imagecustomizerlib/main_test.go) that sets up a temporary directory for the test code. It would probably be goodness to use it here as well.

Reply:
Added this change in.